File: Set1_1000/13_Roman_To_Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romans = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        result = 0
        for i in range(len(s)):
            if i > 0 and romans[s[i]] > romans[s[i-1]]:
                result += romans[s[i]] - 2 * romans[s[i-1]]
            else:
                result += romans[s[i]]
        return result

File: Set1_1000/13_Roman_To_Integer.java
class Solution {
    public int romanToInt(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        int result = 0;
        for (int i=s.length()-1; i>=0; i--) {
            char ch = s.charAt(i);
            int value = 0;
            switch(ch) {
                case 'I' : value = 1;break;
                case 'V' : value = 5;break;
                case 'X' : value = 10;break;
                case 'L' : value = 50;break;
                case 'C' : value = 100;break;
                case 'D' : value = 500;break;
                case 'M' : value = 1000;break; 
            }
            if (4 * value < result) {
                result -= value;
            }
            else {
                result += value;
            }
            // we can clear that by multiplying num by any number between 2 and 4 before comparing it to ans, since the numerals jump in value by increments of at least 5x.


        }
        return result;
    }
}